docs/design/design_v1.8.md

```diff
@@ -0,0 +1,437 @@
+- if authorization mode is WebHook, [Error] if webhook_authz.conf does not exixsts
+
+Please note that:
+
+1. Preflight checks can be invoked individually with the `kubeadm phase preflight` command.
+
+### Generate the necessary certificates
+
+`kubeadm` generates certificate and private key pairs for different purposes.
+Certificates are stored by default in `/etc/kubernetes/pki`. This directory is configurable.
+
+There should be:
+ - a CA certificate (`ca.crt`) with its private key (`ca.key`)
+ - an API Server certificate (`apiserver.crt`) using `ca.crt` as the CA with its private key (`apiserver.key`). The certificate should:
+   - be a serving server certificate (`x509.ExtKeyUsageServerAuth`)
+   - contain altnames for
+     - the kubernetes' service's internal clusterIP and dns name (e.g. `10.96.0.1`, `kubernetes.default.svc.cluster.local`, `kubernetes.default.svc`, `kubernetes.default`, `kubernetes`)
+     - the node-name
+       - **TODO:** I guess this might be a requested feature in opinionated setups, but might be a no-no in more advanced setups. Consensus here?
+     - the IPv4 address of the default route
+     - optional extra altnames that can be specified by the user
+ - a client certificate for the apiservers to connect to the kubelets securely (`apiserver-kubelet-client.crt`) using `ca.crt` as the CA with its private key (`apiserver-kubelet-client.key`). The certificate should:
+   - be a client certificate (`x509.ExtKeyUsageClientAuth`)
+   - be in the `system:masters` organization
+ - a private key for signing ServiceAccount Tokens (`sa.key`) along with its public key (`sa.pub`)
+ - a CA for the front proxy (`front-proxy-ca.crt`) with its key (`front-proxy-ca.key`)
+ - a client cert for the front proxy client (`front-proxy-client.crt`) using `front-proxy-ca.crt` as the CA with its key (`front-proxy-client.key`)
+
+
+Please note that:
+
+1. If a given certificate and private key pair both exist, and its content is evaluated compliant with the above specs, the existing files will be used and the generation phase for the given certificate skipped.
+   This means the user can, for example, copy an existing CA to `/etc/kubernetes/pki/ca.{crt,key}` , and then  then `kubeadm` will use those files for signing the rest of the certs.
+2. Only for the CA, it is possible to provide the `ca.crt` file but not the `ca.key` file, if all other certificates and kubeconfig files already are in place `kubeadm` recognise this condition and activates the so called "ExternalCA" mode, which also implies the `csrsigner`controller in controller-manager won't be started.
+3. If`kubeadm` is running in "ExternalCA" mode; all the certificates must be provided as well, because  `kubeadm` cannot generate them by itself.
+4. In case of `kubeadm`  executed in the `--dry-run` mode, certificates files are written in a temporary folder.
+5. Certificate generation can be invoked individually with the `kubeadm phase certs` command.
```

Comment:
Can we add in every section the command `kubeadm init` runs?
Here, it'd be `kubeadm alpha phase certs all`, etc